resources/views/help/developer.twig

```diff
@@ -38,66 +38,94 @@ Strings that aren't in the new <i>en.catkeys</i> anymore get removed (together w
+
+<li><p><b>Java properties</b> are used by Java projects to assign a translatable text to a specially named variable. As an example, see an export of the WebMessages file of <i>HaikuDepotServer</i> here at Polyglot.</p></li>
+</ul>
+
 <h2 class="help">
 <a href="#">{{ html_image('images/up.png', 'Go up', {'class': 'go-up'}) }}</a>
 <a id="adding" name="adding"></a>Adding a project</h2>
 
 <p>Simply enter the new project name into the <i>Add project</i> text box on the Polyglot start page, click <i>Add</i> and it'll appear in the list of projects. Click on it and you get to the <i>Translation modules</i> page of your project:</p>
 
-<p>{{ html_image('images/help/dev-files-name.png', 'The "Translation modules" page to name parts of your application that need their catkeys translated') }}</p>
+<p>{{ html_image('images/help/dev-files-name.png', 'The "Translation modules" page to name parts of your application that need translation.') }}</p>
+
+<p>Here you enter the name of the translation module you want to add. If it's for the main part of your app it should be called simply "Application". Then you choose the kind of file (Haiku catkeys, line-separated, or Java properties) from the pop-up menu and click on "Add".<p>
 
-<p>After adding the translation module for this main part of your app (by default named "Application"), you upload the <i>en.catkeys</i> for it:</p>
+<p>You arrive at a page to upload the <i>English original</i> for the added module (here for a catkeys module):</p>
 
 <p>{{ html_image('images/help/dev-upload-en-catkeys.png', 'Uploading the en.catkeys') }}</p>
 
-<p>Below that you'll see the name of the translation module, the signature of the application and the checksum which changes every time you create a new <i>en.catkeys</i> with changed GUI strings:</p>
+<p>After you've uploaded a file, you'll see for all kind of files:</p>
+<p>{{ html_image('images/help/dev-catkeys-properties.png', 'Uploading the en.catkeys') }}</p>
+<ul>
+<li><i>Name</i> - the name of the translation module.</li>
+<li><i>Path</i> - the path that'll be used in the ZIP archive that's generated when exporting all languages.</li>
+</ul>
+
+<p>Additionally for Haiku catkeys:</p>
+<ul>
+<li><i>Signature</i> - the signature of the application, as read from the uloaded en.catkeys file.</li>
+<li><i>Checksum</i> - the checksum which changes every time you create a new <i>en.catkeys</i> with changed GUI strings.</li>
+</ul>
 
-<p>{{ html_image('images/help/dev-name-sig-checksum.png', 'The data of a translation module') }}</p>
+<p>Additionally for line-separated files:</p>
+<p>{{ html_image('images/help/dev-upload-line-separated.png', 'Uploading the line-separated file') }}</p>
+<ul>
+<li><i>Extension</i> - the file extension of the uploaded file.</li>
+<li><i>Separator</i> - the characters used to separate translation blocks. Leave empty to use an empty line.</li>
+<li><i>Last key empty</i> - ??????????</li>
```

Comment:
This is internal variable, it keeps track of last line break (or separator) - in Tipster .txt is ended with "%" and no text after - parser reads that as empty key.